src/error.rs

```diff
@@ -8,7 +8,8 @@ pub enum Error {
     Timeout,
     #[error("ecc error {:?}", .0)]
     Ecc(crate::command::EccError),
-
+    #[error("serial port error")]
+    SerialPort(#[from] serialport::Error),
```

Comment:
1. Does this not need a cfg conditional, since that crate is only used when SWI is active?
2. To avoid needing the cfg condition you could translate the error into an IoError?

Reply:
Oh SWI Is no optional anymore, I see